resources/js/components/ui/Table/TQueryTable.tsx

```diff
@@ -29,11 +30,16 @@ import {
   getHeaders,
   tableStyle as ts,
 } from ".";
-import {ColumnFilterController, Spinner} from "..";
+import {ColumnFilterController, FilteringParams, Spinner} from "..";
 
-/** Type of tquery-related information in column meta. */
-export interface TQueryColumnMeta {
-  type: ColumnType;
+export interface ColumnOptions {
+  columnDef?: Partial<IdentifiedColumnDef<object>>;
+  metaParams?: ColumnMetaParams;
+}
+
+interface ColumnMetaParams {
+  canControlVisibility?: boolean;
+  filtering?: FilteringParams;
```

Comment:
trochę boli, że nie ma jakiejś spójności w nazewnictwie, ale to nie jest priorytetem. Jak przyjdzie więcej feature'ów to się wyklaruje